macro/src/wasm_loader.rs

```diff
@@ -39,17 +39,50 @@ fn call_and_decode(wasm_file: Vec<u8>) -> WasmMetadataResult<Metadata> {
 
     let runtime_blob =
         RuntimeBlob::new(&wasm_file).map_err(|e| CodegenError::Wasm(e.to_string()))?;
-    let metadata_encoded = executor
-        .uncached_call(runtime_blob, &mut ext, true, "Metadata_metadata", &[])
-        .map_err(|_| CodegenError::Wasm("method \"Metadata_metadata\" doesnt exist".to_owned()))?;
 
-    let metadata = <Vec<u8>>::decode(&mut &metadata_encoded[..]).map_err(CodegenError::Decode)?;
+    let version = executor
+        .uncached_call(
+            runtime_blob.clone(),
+            &mut ext,
+            true,
+            "Metadata_metadata_versions",
+            &[],
+        )
+        .map_err(|_| {
+            CodegenError::Wasm("method \"Metadata_metadata_versions\" doesnt exist".to_owned())
+        })?;
+    let mut versions = <Vec<u32>>::decode(&mut &version[..]).map_err(CodegenError::Decode)?;
 
-    decode(metadata)
+    // Highest version will always be the last one in the vec
+    versions.sort();
+
+    let version = versions
+        .last()
+        .ok_or(CodegenError::Other(
+            "No metadata versions were returned".to_owned(),
+        ))
+        .map(|v| v.encode())?;
+
+    let encoded_metadata = executor
+        .uncached_call(
+            runtime_blob,
+            &mut ext,
+            false,
+            "Metadata_metadata_at_version",
+            &version,
+        )
+        .map_err(|e| {
+            dbg!(e);
```

Comment:
Could we use tracing:: or log:: here instead of dbg?

Reply:
Oh, I forgot to remove this